aptos-move/block-executor/src/executor.rs

```diff
@@ -601,11 +594,14 @@ where
                     );
                 }
 
-                if scheduler.halt() {
+                if scheduler.halt_ok() {
                     block_limit_processor.finish_parallel_update_counters_and_log_info(
                         txn_idx + 1,
                         scheduler.num_txns(),
                     );
+                    fail_point!("commit-all-halt-err", |_| Err(PanicOr::Or(
+                        IntentionalFallbackToSequential::ResourceGroupSerializationError("".into())
+                    )));
                 }
                 break;
```

Comment:
I know we have Ok(()) after while commit loop terminates, but this technically should be a return?